### ISSUE TITLE
[Windows] Adding desktop library with `add_subdirectory` causes build process to fail

### ISSUE THREAD

**Issue by bilek993** [flutter/flutter#95530]
Hello,

I'm having difficulties with adding any native library into CMake on the Windows platform. Building the project always fails with weird errors. The documentation is non-existent for this topic, so it's really hard to solve this.

## Steps to Reproduce

1. Create a new Flutter project with support for the Windows platform.
2. Copy any CMake C++ library into the project root folder.
3. Add the library in the `[project]\windows\CMakeLists.txt` by adding library subdirectory instruction (`add_subdirectory(../[library_name] "${CMAKE_BINARY_DIR}/${PROJECT_NAME}_directxtex")`).
4. Try to build it in verbose mode (`flutter build windows -v`).

**Expected results:**

It should build without errors.

**Actual results:**

Building application fails when compiling the project.

<details>
<summary>Build logs</summary>

```
[  +40 ms] executing: [C:\Flutter SDK/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +44 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 18116933e77adc82f80866c928266a5b4f1ed645
[        ] executing: [C:\Flutter SDK/] git tag --points-at 18116933e77adc82f80866c928266a5b4f1ed645
[  +31 ms] Exit code 0 from: git tag --points-at 18116933e77adc82f80866c928266a5b4f1ed645
[        ] 2.5.3
[   +5 ms] executing: [C:\Flutter SDK/] git rev-parse --abbrev-ref --symbolic @{u}
[  +25 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [C:\Flutter SDK/] git ls-remote --get-url origin
[  +23 ms] Exit code 0 from: git ls-remote --get-url origin
[   +5 ms] https://github.com/flutter/flutter.git
[  +76 ms] executing: [C:\Flutter SDK/] git rev-parse --abbrev-ref HEAD
[  +26 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +46 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +34 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +29 ms] Skipping pub get: version match.
[  +80 ms]  Building with sound null safety 
[   +6 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[  +23 ms] Exit code 0 from: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[        ] [
             {
               "instanceId": "1de8f7d4",
               "installDate": "2021-11-08T20:33:38Z",
               "installationName": "VisualStudio/16.11.6+31829.152",
               "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community",
               "installationVersion": "16.11.31829.152",
               "productId": "Microsoft.VisualStudio.Product.Community",
               "productPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\Common7\\IDE\\devenv.exe",
               "state": 4294967295,
               "isComplete": true,
               "isLaunchable": true,
               "isPrerelease": false,
               "isRebootRequired": false,
               "displayName": "Visual Studio Community 2019",
               "description": "Zaawansowane środowisko IDE — bezpłatne dla uczniów i studentów, współautorów oprogramowania open source oraz indywidualnych osób",
               "channelId": "VisualStudio.16.Release",
               "channelUri": "https://aka.ms/vs/16/release/channel",
               "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
               "releaseNotes": "https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-v16.11#16.11.6",
               "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=660909",
               "updateDate": "2021-11-09T19:29:55.7541279Z",
               "catalog": {
                 "buildBranch": "d16.11",
                 "buildVersion": "16.11.31829.152",
                 "id": "VisualStudio/16.11.6+31829.152",
                 "localBuild": "build-lab",
                 "manifestName": "VisualStudio",
                 "manifestType": "installer",
                 "productDisplayVersion": "16.11.6",
                 "productLine": "Dev16",
                 "productLineVersion": "2019",
                 "productMilestone": "RTW",
                 "productMilestoneIsPreRelease": "False",
                 "productName": "Visual Studio",
                 "productPatchVersion": "6",
                 "productPreReleaseMilestoneSuffix": "1.0",
                 "productSemanticVersion": "16.11.6+31829.152",
                 "requiredEngineVersion": "2.11.47.9733"
               },
               "properties": {
                 "campaignId": "",
                 "channelManifestId": "VisualStudio.16.Release/16.11.6+31829.152",
                 "nickname": "",
                 "setupEngineFilePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\setup.exe"
               }
             }
           ]
[   +2 ms] Building Windows application...
[   +9 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe -S C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\windows -B build\windows -G Visual Studio 16 2019
[  +29 ms] -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.22000.
[  +24 ms] -- Configuring done
[  +94 ms] -- Generating done
[   +3 ms] -- Build files have been written to: C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/build/windows
[   +5 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe --build build\windows --config Release --target INSTALL --verbose
[  +96 ms] Microsoft (R) Build Engine 16.11.2+f32259642 dla platformy .NET Framework
[        ] Copyright (C) Microsoft Corporation. Wszelkie prawa zastrzeżone.
[  +35 ms] Kompilacja rozpoczęła się 19.12.2021 20:01:26.
[ +168 ms] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\INSTALL.vcxproj” w węźle 1 (domyślne elementy docelowe).
[        ] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\INSTALL.vcxproj” (1) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ZERO_CHECK.vcxproj” (2) w węźle 1 (domyślne elementy docelowe).
[        ] InitializeBuildStatus:
[        ]   Trwa tworzenie „x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[  +19 ms] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[  +68 ms]   Checking Build System
[  +48 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ZERO_CHECK.vcxproj” wykonane (domyślne elementy docelowe).
[  +50 ms] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\INSTALL.vcxproj” (1) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” (3) w węźle 1 (domyślne elementy docelowe).
[        ] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” (3) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\stackoverflow_windows_question_directxtex\DirectXTex.vcxproj” (4) w węźle 1 (domyślne elementy docelowe).
[        ] InitializeBuildStatus:
[        ]   Trwa tworzenie „DirectXTex.dir\Release\DirectXTex.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[   +1 ms] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[  +30 ms] ClCompile:
[        ]   Wszystkie dane wyjściowe są aktualne.
[  +10 ms]   Wszystkie dane wyjściowe są aktualne.
[   +9 ms] Lib:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +1 ms]   DirectXTex.vcxproj -> C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\bin\CMake\Release\DirectXTex.lib
[   +8 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „DirectXTex.dir\Release\DirectXTex.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „DirectXTex.dir\Release\DirectXTex.tlog\DirectXTex.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\stackoverflow_windows_question_directxtex\DirectXTex.vcxproj” wykonane (domyślne elementy docelowe).
[  +29 ms] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” (3) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\flutter_wrapper_app.vcxproj” (5) w węźle 1 (domyślne elementy docelowe).
[        ] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\flutter_wrapper_app.vcxproj” (5) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\flutter_assemble.vcxproj” (6) w węźle 1 (domyślne elementy docelowe).
[        ] InitializeBuildStatus:
[        ]   Trwa tworzenie „x64\Release\flutter_assemble\flutter_assemble.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[   +4 ms] CustomBuild:
[        ]   "The build of 'C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\CMakeFiles\e44387c17a1afe4a6b4e18e8a857a4dd\flutter_assemble.rule' depends on 'C:\USERS\JBILI\DOCUMENTS\FLUTTER-PROJECTS\STACKOVERFLOW_WINDOWS_QUESTION\WINDOWS\FLUTTER\EPHEMERAL\FLUTTER_WINDOWS.DLL' which is produced by the build of 'C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\CMakeFiles\6bdc379a9663a8edd000a236390c7888\flutter_windows.dll.rule'. The items cannot be built in parallel."
[  +47 ms]   Generating C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/flutter_windows.dll, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/flutter_export.h, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/flutter_windows.h, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/flutter_messenger.h, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/flutter_plugin_registrar.h, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/flutter_texture_registrar.h, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/cpp_client_wrapper/core_implementations.cc, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/cpp_client_wrapper/standard_codec.cc, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/cpp_client_wrapper/plugin_registrar.cc, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/cpp_client_wrapper/flutter_engine.cc, C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/flutter/ephemeral/cpp_client_wrapper/flutter_view_controller.cc, _phony_
[ +547 ms]   [  +40 ms] executing: [C:\Flutter SDK/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +35 ms]   [  +48 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ]   [        ] 18116933e77adc82f80866c928266a5b4f1ed645
[        ]   [        ] executing: [C:\Flutter SDK/] git tag --points-at 18116933e77adc82f80866c928266a5b4f1ed645
[  +35 ms]   [  +35 ms] Exit code 0 from: git tag --points-at 18116933e77adc82f80866c928266a5b4f1ed645
[        ]   [        ] 2.5.3
[   +5 ms]   [   +5 ms] executing: [C:\Flutter SDK/] git rev-parse --abbrev-ref --symbolic @{u}
[  +30 ms]   [  +30 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ]   [        ] origin/stable
[        ]   [        ] executing: [C:\Flutter SDK/] git ls-remote --get-url origin
[  +28 ms]   [  +28 ms] Exit code 0 from: git ls-remote --get-url origin
[        ]   [        ] https://github.com/flutter/flutter.git
[  +37 ms]   [  +37 ms] executing: [C:\Flutter SDK/] git rev-parse --abbrev-ref HEAD
[  +29 ms]   [  +29 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ]   [        ] stable
[  +40 ms]   [  +40 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms]   [   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +67 ms]   [  +67 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[   +1 ms]   [   +1 ms] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +19 ms]   [  +20 ms] Initializing file store
[  +10 ms]   [   +9 ms] Done initializing file store
[  +34 ms]   [  +34 ms] Skipping target: gen_localizations
[   +6 ms]   [   +6 ms] Skipping target: gen_dart_plugin_registrant
[+1315 ms]   [+1314 ms] Skipping target: unpack_windows
[ +255 ms]   [ +256 ms] Skipping target: kernel_snapshot
[  +13 ms]   [  +13 ms] Skipping target: aot_elf_release
[  +11 ms]   [  +11 ms] Skipping target: windows_aot_bundle
[  +20 ms]   [  +20 ms] Skipping target: release_bundle_windows_assets
[        ]   [        ] Persisting file store
[   +6 ms]   [   +6 ms] Done persisting file store
[   +2 ms]   [   +2 ms] build succeeded.
[   +5 ms]   [   +5 ms] "flutter assemble" took 1┬á786ms.
[ +110 ms]   [ +110 ms] ensureAnalyticsSent: 108ms
[   +1 ms]   [   +1 ms] Running shutdown hooks
[        ]   [        ] Shutdown hooks complete
[        ]   [        ] exiting with code 0
[  +49 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „x64\Release\flutter_assemble\flutter_assemble.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „x64\Release\flutter_assemble\flutter_assemble.tlog\flutter_assemble.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\flutter_assemble.vcxproj” wykonane (domyślne elementy docelowe).
[   +4 ms] InitializeBuildStatus:
[        ]   Trwa tworzenie „flutter_wrapper_app.dir\Release\flutter_.68EC41A2.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[   +1 ms] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +7 ms] ClCompile:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +3 ms] Lib:
[        ]   Wszystkie dane wyjściowe są aktualne.
[        ]   flutter_wrapper_app.vcxproj -> C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\Release\flutter_wrapper_app.lib
[   +6 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „flutter_wrapper_app.dir\Release\flutter_.68EC41A2.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „flutter_wrapper_app.dir\Release\flutter_.68EC41A2.tlog\flutter_wrapper_app.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\flutter_wrapper_app.vcxproj” wykonane (domyślne elementy docelowe).
[  +25 ms] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” (3) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\flutter_wrapper_plugin.vcxproj” (7) w węźle 1 (domyślne elementy docelowe).
[        ] InitializeBuildStatus:
[        ]   Trwa tworzenie „flutter_wrapper_plugin.dir\Release\flutter_.0F0012E8.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[   +4 ms] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +6 ms] ClCompile:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +2 ms] Lib:
[        ]   Wszystkie dane wyjściowe są aktualne.
[        ]   flutter_wrapper_plugin.vcxproj -> C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\Release\flutter_wrapper_plugin.lib
[   +3 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „flutter_wrapper_plugin.dir\Release\flutter_.0F0012E8.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „flutter_wrapper_plugin.dir\Release\flutter_.0F0012E8.tlog\flutter_wrapper_plugin.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\flutter\flutter_wrapper_plugin.vcxproj” wykonane (domyślne elementy docelowe).
[  +22 ms] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” (3) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\runner\stackoverflow_windows_question.vcxproj” (8) w węźle 1 (domyślne elementy docelowe).
[        ] InitializeBuildStatus:
[        ]   Trwa tworzenie „stackoverflow_windows_question.dir\Release\stackove.2EF32D47.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[   +1 ms] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[  +13 ms] ClCompile:
[        ]   C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\CL.exe /c /I"C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\windows" /I"C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\windows\flutter\ephemeral" /I"C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\windows\flutter\ephemeral\cpp_client_wrapper\include" /nologo /W4 /WX /diagnostics:column /O2 /Ob2 /D WIN32 /D _WINDOWS /D NDEBUG /D _HAS_EXCEPTIONS=0 /D NOMINMAX /D UNICODE /D _UNICODE /D "CMAKE_INTDIR=\"Release\"" /D _UNICODE /D UNICODE /Gm- /EHsc /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++17 /Fo"stackoverflow_windows_question.dir\Release\\" /Fd"stackoverflow_windows_question.dir\Release\vc142.pdb" /external:W4 /Gd /TP /wd4100 /errorReport:queue "C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\windows\flutter\generated_plugin_registrant.cc" "C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\windows\runner\flutter_window.cpp"
[  +67 ms]   generated_plugin_registrant.cc
[ +235 ms]   flutter_window.cpp
[ +444 ms]   Generating Code...
[  +79 ms] ResourceCompile:
[        ]   Wszystkie dane wyjściowe są aktualne.
[  +18 ms] Link:
[        ]   C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\link.exe /ERRORREPORT:QUEUE /OUT:"C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\runner\Release\stackoverflow_windows_question.exe" /INCREMENTAL:NO /NOLOGO ..\flutter\Release\flutter_wrapper_app.lib "C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\windows\flutter\ephemeral\flutter_windows.dll.lib" kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /manifestinput:"C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\windows\runner\runner.exe.manifest" /PDB:"C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/build/windows/runner/Release/stackoverflow_windows_question.pdb" /SUBSYSTEM:WINDOWS /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/build/windows/runner/Release/stackoverflow_windows_question.lib" /MACHINE:X64  /machine:x64 stackoverflow_windows_question.dir\Release\Runner.res
[        ]   stackoverflow_windows_question.dir\Release\flutter_window.obj
[        ]   stackoverflow_windows_question.dir\Release\main.obj
[        ]   stackoverflow_windows_question.dir\Release\utils.obj
[        ]   stackoverflow_windows_question.dir\Release\win32_window.obj
[        ]   stackoverflow_windows_question.dir\Release\generated_plugin_registrant.obj
[ +154 ms]   stackoverflow_windows_question.vcxproj -> C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\runner\Release\stackoverflow_windows_question.exe
[   +3 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „stackoverflow_windows_question.dir\Release\stackove.2EF32D47.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „stackoverflow_windows_question.dir\Release\stackove.2EF32D47.tlog\stackoverflow_windows_question.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\runner\stackoverflow_windows_question.vcxproj” wykonane (domyślne elementy docelowe).
[  +19 ms] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” (3) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\stackoverflow_windows_question_directxtex\texassemble.vcxproj” (9) w węźle 1 (domyślne elementy docelowe).
[        ] InitializeBuildStatus:
[        ]   Trwa tworzenie „texassemble.dir\Release\texassemble.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[   +1 ms] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +5 ms] ClCompile:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +9 ms] Link:
[        ]   Wszystkie dane wyjściowe są aktualne.
[        ]   texassemble.vcxproj -> C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\bin\CMake\Release\texassemble.exe
[   +3 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „texassemble.dir\Release\texassemble.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „texassemble.dir\Release\texassemble.tlog\texassemble.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\stackoverflow_windows_question_directxtex\texassemble.vcxproj” wykonane (domyślne elementy docelowe).
[  +20 ms] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” (3) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\stackoverflow_windows_question_directxtex\texconv.vcxproj” (10) w węźle 1 (domyślne elementy docelowe).
[        ] InitializeBuildStatus:
[        ]   Trwa tworzenie „texconv.dir\Release\texconv.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[   +3 ms] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +6 ms] ClCompile:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +6 ms] Link:
[        ]   Wszystkie dane wyjściowe są aktualne.
[        ]   texconv.vcxproj -> C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\bin\CMake\Release\texconv.exe
[   +3 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „texconv.dir\Release\texconv.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „texconv.dir\Release\texconv.tlog\texconv.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\stackoverflow_windows_question_directxtex\texconv.vcxproj” wykonane (domyślne elementy docelowe).
[  +22 ms] Projekt „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” (3) kompiluje „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\stackoverflow_windows_question_directxtex\texdiag.vcxproj” (11) w węźle 1 (domyślne elementy docelowe).
[        ] InitializeBuildStatus:
[        ]   Trwa tworzenie „texdiag.dir\Release\texdiag.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[        ] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +4 ms] ClCompile:
[        ]   Wszystkie dane wyjściowe są aktualne.
[   +8 ms] Link:
[        ]   Wszystkie dane wyjściowe są aktualne.
[        ]   texdiag.vcxproj -> C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\bin\CMake\Release\texdiag.exe
[   +3 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „texdiag.dir\Release\texdiag.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „texdiag.dir\Release\texdiag.tlog\texdiag.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\stackoverflow_windows_question_directxtex\texdiag.vcxproj” wykonane (domyślne elementy docelowe).
[   +6 ms] InitializeBuildStatus:
[        ]   Trwa tworzenie „x64\Release\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild”, ponieważ określono "AlwaysCreate".
[   +2 ms] CustomBuild:
[        ]   Wszystkie dane wyjściowe są aktualne.
[  +42 ms]   Building Custom Rule C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/windows/CMakeLists.txt
[  +32 ms] FinalizeBuildStatus:
[        ]   Usuwanie pliku „x64\Release\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild”.
[        ]   Modyfikowanie „x64\Release\ALL_BUILD\ALL_BUILD.tlog\ALL_BUILD.lastbuildstate”.
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\ALL_BUILD.vcxproj” wykonane (domyślne elementy docelowe).
[   +3 ms] InitializeBuildStatus:
[        ]   Modyfikowanie „x64\Release\INSTALL\INSTALL.tlog\unsuccessfulbuild”.
[  +54 ms] PostBuildEvent:
[        ]   setlocal
[        ]   "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DBUILD_TYPE=Release -P cmake_install.cmake
[        ]   if %errorlevel% neq 0 goto :cmEnd
[        ]   :cmEnd
[        ]   endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
[        ]   :cmErrorLevel
[        ]   exit /b %1
[        ]   :cmDone
[        ]   if %errorlevel% neq 0 goto :VCEnd
[        ]   :VCEnd
[  +57 ms]   -- Install configuration: "Release"
[        ]   CMake Error at stackoverflow_windows_question_directxtex/cmake_install.cmake:41 (file):
[        ]     file cannot create directory:
[        ]     C:/Users/jbili/Documents/Flutter-Projects/stackoverflow_windows_question/build/windows/$<TARGET_FILE_DIR:stackoverflow_windows_question>/lib.
[        ]     Maybe need administrative privileges.
[        ]   Call Stack (most recent call first):
[        ]     cmake_install.cmake:47 (include)
[        ] Kompilowanie projektu „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\INSTALL.vcxproj” wykonane (domyślne elementy docelowe) — NIEPOWODZENIE.
[        ] Kompilacja NIE POWIODŁA SIĘ.
[        ] „C:\Users\jbili\Documents\Flutter-Projects\stackoverflow_windows_question\build\windows\INSTALL.vcxproj” (domyślny element docelowy) (1)->
[        ] (element docelowy PostBuildEvent) -> 
[        ]     Ostrzeżenia: 0
[        ]     Liczba błędów: 1
[        ] Czas, który upłynął: 00:00:04.67
[  +16 ms] Building Windows application... (completed in 5,0s)
[        ] "flutter windows" took 5 195ms.
[  +30 ms] ensureAnalyticsSent: 29ms
[        ] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```

</details>

<details>
<summary>Flutter Doctor</summary>

```
[√] Flutter (Channel stable, 2.5.3, on Microsoft Windows [Version 10.0.22000.376], locale pl-PL)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.6)
[√] Android Studio (version 2020.3)
[√] IntelliJ IDEA Ultimate Edition (version 2021.2)
[√] VS Code (version 1.63.0)
[√] Connected device (3 available)
```

</details>

<details>
<summary>CMakeLists.txt</summary>

```
cmake_minimum_required(VERSION 3.15)
project(stackoverflow_windows_question LANGUAGES CXX)

set(BINARY_NAME "stackoverflow_windows_question")

cmake_policy(SET CMP0063 NEW)

set(CMAKE_INSTALL_RPATH "$ORIGIN/lib")

# Configure build options.
get_property(IS_MULTICONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
if(IS_MULTICONFIG)
  set(CMAKE_CONFIGURATION_TYPES "Debug;Profile;Release"
    CACHE STRING "" FORCE)
else()
  if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
    set(CMAKE_BUILD_TYPE "Debug" CACHE
      STRING "Flutter build mode" FORCE)
    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
      "Debug" "Profile" "Release")
  endif()
endif()

set(CMAKE_EXE_LINKER_FLAGS_PROFILE "${CMAKE_EXE_LINKER_FLAGS_RELEASE}")
set(CMAKE_SHARED_LINKER_FLAGS_PROFILE "${CMAKE_SHARED_LINKER_FLAGS_RELEASE}")
set(CMAKE_C_FLAGS_PROFILE "${CMAKE_C_FLAGS_RELEASE}")
set(CMAKE_CXX_FLAGS_PROFILE "${CMAKE_CXX_FLAGS_RELEASE}")

# Use Unicode for all projects.
add_definitions(-DUNICODE -D_UNICODE)

# Compilation settings that should be applied to most targets.
function(APPLY_STANDARD_SETTINGS TARGET)
  target_compile_features(${TARGET} PUBLIC cxx_std_17)
  target_compile_options(${TARGET} PRIVATE /W4 /WX /wd"4100")
  target_compile_options(${TARGET} PRIVATE /EHsc)
  target_compile_definitions(${TARGET} PRIVATE "_HAS_EXCEPTIONS=0")
  target_compile_definitions(${TARGET} PRIVATE "$<$<CONFIG:Debug>:_DEBUG>")
endfunction()

set(FLUTTER_MANAGED_DIR "${CMAKE_CURRENT_SOURCE_DIR}/flutter")

# Flutter library and tool build rules.
add_subdirectory(${FLUTTER_MANAGED_DIR})

# Application build
add_subdirectory("runner")

# Generated plugin build rules, which manage building the plugins and adding
# them to the application.
include(flutter/generated_plugins.cmake)

add_subdirectory(../DirectXTex "${CMAKE_BINARY_DIR}/${PROJECT_NAME}_directxtex") # NEW LINE

# === Installation ===
# Support files are copied into place next to the executable, so that it can
# run in place. This is done instead of making a separate bundle (as on Linux)
# so that building and running from within Visual Studio will work.
set(BUILD_BUNDLE_DIR "$<TARGET_FILE_DIR:${BINARY_NAME}>")
# Make the "install" step default, as it's required to run.
set(CMAKE_VS_INCLUDE_INSTALL_TO_DEFAULT_BUILD 1)
if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
  set(CMAKE_INSTALL_PREFIX "${BUILD_BUNDLE_DIR}" CACHE PATH "..." FORCE)
endif()

set(INSTALL_BUNDLE_DATA_DIR "${CMAKE_INSTALL_PREFIX}/data")
set(INSTALL_BUNDLE_LIB_DIR "${CMAKE_INSTALL_PREFIX}")

install(TARGETS ${BINARY_NAME} RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}"
  COMPONENT Runtime)

install(FILES "${FLUTTER_ICU_DATA_FILE}" DESTINATION "${INSTALL_BUNDLE_DATA_DIR}"
  COMPONENT Runtime)

install(FILES "${FLUTTER_LIBRARY}" DESTINATION "${INSTALL_BUNDLE_LIB_DIR}"
  COMPONENT Runtime)

if(PLUGIN_BUNDLED_LIBRARIES)
  install(FILES "${PLUGIN_BUNDLED_LIBRARIES}"
    DESTINATION "${INSTALL_BUNDLE_LIB_DIR}"
    COMPONENT Runtime)
endif()

# Fully re-copy the assets directory on each build to avoid having stale files
# from a previous install.
set(FLUTTER_ASSET_DIR_NAME "flutter_assets")
install(CODE "
  file(REMOVE_RECURSE \"${INSTALL_BUNDLE_DATA_DIR}/${FLUTTER_ASSET_DIR_NAME}\")
  " COMPONENT Runtime)
install(DIRECTORY "${PROJECT_BUILD_DIR}/${FLUTTER_ASSET_DIR_NAME}"
  DESTINATION "${INSTALL_BUNDLE_DATA_DIR}" COMPONENT Runtime)

# Install the AOT library on non-Debug builds only.
install(FILES "${AOT_LIBRARY}" DESTINATION "${INSTALL_BUNDLE_DATA_DIR}"
  CONFIGURATIONS Profile;Release
  COMPONENT Runtime)
  ```

</details>

**Comment by stuartmorgan** [comment]
> I'm having difficulties with adding any native library

Could you provide a minimal reproducible example? This looks like it's probably some kind of interaction with `CMAKE_INSTALL_PREFIX`, so it seems unlikely that it's *any* library; it's likely specific to certain details of the library's CMake build.

**Comment by bilek993** [comment]
@stuartmorgan 
Yes, I was trying to add [DirectXTex](https://github.com/microsoft/DirectXTex) library. Like I wrote in the reproduction steps, you can just copy it to the root project directory and use `CMakeLists.txt` that I provided because `add_subdirectory(../DirectXTex "${CMAKE_BINARY_DIR}/${PROJECT_NAME}_directxtex")` is pointing to this library.

**Comment by maheshmnj** [comment]
Hi @bilek993, thanks for filing the issue I tried the above reproducible steps on stable and the master channel and I was able to reproduce the same behavior the desktop app fails to build or run after adding the library in `CMakeLists.txt` file.
Leaving this issue open for further insights from the team.

<details>
<summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 2.9.0-1.0.pre.95, on Microsoft Windows [Version 10.0.19042.1348],    
    locale en-US)
    • Flutter version 2.9.0-1.0.pre.95 at C:\flutter_sdk\master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 935d1478aa (5 days ago), 2021-12-14 23:04:05 -0800
    • Engine revision 57f7719e6e
    • Dart version 2.16.0 (build 2.16.0-109.0.dev)
    • DevTools version 2.9.1

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)    
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)      
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.61.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code   
    • Flutter extension version 3.29.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1348]  
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 96.0.4664.110
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 96.0.1054.53

• No issues found!
```

```
[√] Flutter (Channel stable, 2.8.0, on Microsoft Windows [Version 10.0.19042.1348], locale en-US)
    • Flutter version 2.8.0 at C:\flutter_sdk\stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cf44000065 (12 days ago), 2021-12-08 14:06:50 -0800
    • Engine revision 40a99c5951
    • Dart version 2.15.0

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)    
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)        
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)      
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.61.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.29.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1348]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 96.0.4664.110
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 96.0.1054.53

• No issues found!
```
</details>  

Here are the complete [logs](https://github.com/flutter/flutter/files/7745891/run_logs.zip) and the [sample project](https://github.com/flutter/flutter/files/7745903/hello.zip)

**Comment by stuartmorgan** [comment]
It would take some investigation to figure out what exactly is happening; it looks like maybe that project is using the install prefix in a context that doesn't support generator expressions. But there's a more structural problem: the Runner build uses `install` in an unusual way, to create the self-contained app bundle (since on Linux at least it means the rpaths are automatically fixed up, I can't remember if there was a specific advantage on Windows). That project's build, on the other hand, has a more standard use of install, which means that when Flutter runs the install step that project would put a bunch of random things inside the bundle. Avoiding that would require a completely different approach to how Flutter's runner build works.

I'm not sure this should actually be considered a bug though. I'm not a CMake expert by any means, but I wouldn't expect including a completely different project's build--written and maintained by different people for a different use case--as a subdirectory would actually work in general. It would have unintended install effects even with more normal use of install, and there are also global CMake settings that a build file can change that would cause unexpected interactions between the two builds. I would expect the normal way of doing this would be to either:
- pre-build the other project and use its libraries, or
- sub-run a completely separate build step as a custom script step, then use the output.

@bilek993 Could you point to sources that recommend this way of using CMake?

**Comment by bilek993** [comment]
@stuartmorgan 
I'm not a CMake expert either, but I think adding libraries as subprojects is one of the ways of handling dependencies, because this way building the main project (in our case Flutter project) should result in rebuilding all of the libraries automatically. We can find this way of handling dependencies (with `add_subdirectory`) on [many websites](https://www.selectiveintellect.net/blog/2016/7/29/using-cmake-to-add-third-party-libraries-to-your-project-1) and even in the [official CMake documentation](https://cmake.org/cmake/help/latest/guide/tutorial/Adding%20a%20Library.html).

**Comment by stuartmorgan** [comment]
> We can find this way of handling dependencies (with `add_subdirectory`) on [many websites](https://www.selectiveintellect.net/blog/2016/7/29/using-cmake-to-add-third-party-libraries-to-your-project-1)

That is using `ExternalProject`, not `add_subdirectories`, which is very different. The only things added by `add_subdirectory` in that example are CMake files **authored by the person writing that project**.

> and even in the [official CMake documentation](https://cmake.org/cmake/help/latest/guide/tutorial/Adding%20a%20Library.html).

As it says in the intro to that page:

> This library will contain **our own implementation** for computing the square root of a number.

Emphasis added. Note that, again, it instructs **you** to write the CMake file. That's simply describing a way of making a project's build modular for maintainability. Flutter's runner does that itself.

Both of these are doing something that is extremely common, and fully supported by the Flutter template's build file: adding **your own** build files which are written for the purpose of being part of that build. What they are **not** doing is simply dropping someone else's CMake build into their own as a component and expecting it to work. Can you point to documents describing that as best practice? That's what I was asking for.

**Comment by dubajj** [comment]
just adding some info here, i have this same issue with the same failure for using `add_subdirectory` with an external cmake project.  the string `$<TARGET_FILE_DIR:....` being passed as part of the path name. Adding `EXCLUDE_FROM_ALL` is a functional workaround for me.

so @bilek993 may be able to get running by changing 
`add_subdirectory(../[library_name] "${CMAKE_BINARY_DIR}/${PROJECT_NAME}_directxtex")`
to
`add_subdirectory(../[library_name] "${CMAKE_BINARY_DIR}/${PROJECT_NAME}_directxtex" EXCLUDE_FROM_ALL)`

**Comment by felixdollack** [comment]
I have the same issue. I see `$<TARGET_FILE_DIR:....` in the generated cmake files but adding the `EXCLUDE_FROM_ALL` option to the add_subdirectory command, as @dubajj described, helps.

My flutter version:
```
Flutter 3.3.8 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 52b3dc25f6 (8 days ago) • 2022-11-09 12:09:26 +0800
Engine • revision 857bd6b74c
Tools • Dart 2.18.4 • DevTools 2.15.0
```

**Comment by stuartmorgan** [comment]
Closing as out of scope; as explained above, I don't see any evidence that combining multiple unrelated projects into the same CMake build is a scenario that CMake is designed to support. External builds should be driven as such, not grafted in.
